python/lsst/cp/pipe/ptc.py

```diff
@@ -1133,3 +1122,35 @@ def errFunc(p, x, y):
             dataset.badAmps = np.repeat(np.nan, len(list(dataset.rawExpTimes.values())[0]))
 
         return dataset
+
+    def fillBadAmp(self, dataset, ptcFitType, ampName):
+        """Fill the dataset with NaNs if there are not enough good points.
+
+        Parameters
+        ----------
+        dataset : `lsst.ip.isr.ptcDataset.PhotonTransferCurveDataset`
+            The dataset containing the means, variances and exposure times.
+
+        ptcFitType : `str`
+            Fit a 'POLYNOMIAL' (degree: 'polynomialFitDegree') or
+            'EXPAPPROXIMATION' (Eq. 16 of Astier+19) to the PTC.
```

Comment:
Can this not happen with FULLCOVARIANCE?

Reply:
This helper function is used by `fitPtc`, which is used only by `POLYNOMIAL` or `EXPAPPROXIMATION`. It was written just to facilitate the reading of `fitPtc`.